### PR TITLE
Syntax kinds service cleanup

### PR DESCRIPTION
**Pull request by petrroll**
Based on #37196 (only last commit is new).

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
da262c1  is good for me.

**Comment by heejaechang** [comment]
test failures?

File: src/Features/Core/Portable/IntroduceUsingStatement/AbstractIntroduceUsingStatementCodeRefactoringProvider.cs [hunk 1]
```diff
@@ -46,11 +46,8 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
 
         private async Task<TLocalDeclarationSyntax> FindDisposableLocalDeclaration(Document document, TextSpan selection, CancellationToken cancellationToken)
         {
-            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
-
-            var declarationSyntax =
-                root.FindNode(selection)?.GetAncestorOrThis<TLocalDeclarationSyntax>()
-                ?? root.FindTokenOnLeftOfPosition(selection.End).GetAncestor<TLocalDeclarationSyntax>();
+            var refactoringHelperService = document.GetLanguageService<IRefactoringHelpersService>();
```
Comment (heejaechang) [inline src/Features/Core/Portable/IntroduceUsingStatement/AbstractIntroduceUsingStatementCodeRefactoringProvider.cs]: this approach still force users to put caret on right location for refactoring to be suggested? @mavasani ?

Comment (petrroll) [reply src/Features/Core/Portable/IntroduceUsingStatement/AbstractIntroduceUsingStatementCodeRefactoringProvider.cs]: If anything this should allow the refactoring for more locations. I was very careful to not to regress (showed before, doesn't show now).

**Comment by petrroll** [comment]
DevOps issues.

**Comment by mavasani** [comment]
@petrroll Checks are running now.

**Pull request merged**